### PR TITLE
Hydra Test Coverage for Complex Data with ~1GB Data + Snappy Smart Connector Mode

### PR DESCRIPTION
**Pull request by BhattChandresh**
## Changes proposed in this pull request

- Hydra Test Coverage for Complex Data Type  with ~1GB Data.
- Hydra Test Coverage for Complex Data Type with Snappy smart connector mode.
- Data Types covered - 
        > ArrayType
        > MapType
        > StructType
        > Combination of above all 3 types in one table (ArrayType,MapType,StructType)
        > ArrayOfStructType
        > ArrayOfStringInMapAsValue
 
## Patch testing
- Changed the dataset from few rows to ~1 GB and re-run the battery suit.

## ReleaseNotes.txt changes

NA

## Other PRs 
None

### CODE REVIEW

**Comment by PradeepSurale** [comment]
Can one of the admins verify this patch?

File: dtests/src/test/java/io/snappydata/hydra/complexdatatypes/arrayTypes.conf [hunk 1]
```diff
@@ -1,14 +1,26 @@
-hydra.Prms-testRequirement = "Test for complex data types - array type, map type";
+hydra.Prms-testRequirement = "Test for complex data types - array type";
```
Comment (sonalsagarwal) [inline dtests/src/test/java/io/snappydata/hydra/complexdatatypes/arrayTypes.conf]: Please modify the test description.

File: dtests/src/test/java/io/snappydata/hydra/complexdatatypes/arrayTypes.conf [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
 
-TASK       taskClass   = io.snappydata.hydra.cluster.SnappyTest taskMethod = HydraTask_executeSnappyJob
+TASK   taskClass = io.snappydata.hydra.cluster.SnappyTest taskMethod = HydraTask_executeSnappyJob
            io.snappydata.hydra.cluster.SnappyPrms-jobClassNames = io.snappydata.hydra.complexdatatypes.ArrayType
            io.snappydata.hydra.cluster.SnappyPrms-userAppName = "ComplexType_ArrayType_Validation"
+           io.snappydata.hydra.cluster.SnappyPrms-appPropsForJobServer ="dataFilesLocation=${dataFilesLocation}"
            io.snappydata.hydra.cluster.SnappyPrms-userAppJar = snappydata-store-scala-tests*tests.jar
            threadGroups = snappyThreads
            maxTimesToRun = 1;
+/*
```
Comment (sonalsagarwal) [inline dtests/src/test/java/io/snappydata/hydra/complexdatatypes/arrayTypes.conf]: Can you please check why this task is commented, please add the reason for it.

File: dtests/src/test/java/io/snappydata/hydra/complexdatatypes/complexDataTypes.bt [hunk 3]
```diff
@@ -1,54 +1,126 @@
...
+  A=snappyStore snappyStoreHosts=3 snappyStoreVMsPerHost=1 snappyStoreThreadsPerVM=1
+  B=lead leadHosts=2 leadVMsPerHost=1 leadThreadsPerVM=2
+  C=locator locatorHosts=3 locatorVMsPerHost=1 locatorThreadsPerVM=1
+  D=worker workerHosts=3 workerVMsPerHost=1 workerThreadsPerVM=1
+  dataFilesLocation="/export/shared/QA_DATA/Complextypes_Data/StructType/"
+  numRowsValidation = true
+  fullResultSetValidation = true
+  isSmokeRun = false
+
+/*
```
Comment (sonalsagarwal) [inline dtests/src/test/java/io/snappydata/hydra/complexdatatypes/complexDataTypes.bt]: Please resolve the commented code.

File: dtests/src/test/java/io/snappydata/hydra/complexdatatypes/arrayTypes.conf (hunk 1 above)
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/complexdatatypes/arrayTypes.conf]: Done.

File: dtests/src/test/java/io/snappydata/hydra/complexdatatypes/complexDataTypes.bt (hunk 3 above)
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/complexdatatypes/complexDataTypes.bt]: Remove the comments. All the tests needs to be in master.

**Pull request merged**